### DISCUSSION TITLE
Strict Mode variables different in different environments?

### DISCUSSION THREAD

**Discussion by ya2s** [vercel/turborepo#8367] [Help]
### Summary

Updated turbo to 2.0.

Some environment variables are no longer loaded.
When I looked at the documentation, I learned that it will not be loaded unless you set it to `env ` or `globalEnv`.

However, some environment variables seemed to be read.
I understand that [framework-specific environment variables](https://turbo.build/repo/docs/crafting-your-repository/using-environment-variables#framework-inference) are read.
It seems that other environment variables are also being read.

Especially when I ran `next build`, the environment variables set in [vercel](https://nextjs.org/docs/pages/building-your-application/configuring/environment-variables#environment-variables-on-vercel) seemed to be loaded without any problems. 
Why is this?
Also, when I run `yarn dev`, it seems that I am able to read the .env environment variables even though they are not set.

turbo.json
```
{
  "$schema": "https://turbo.build/schema.json",
  "globalEnv": ["NODE_ENV"],
  "tasks": {
    "build": {
      "outputs": [".next/**", "dist/**", "build/**"],
      "dependsOn": ["^build"]
    },
    "dev": {
      "cache": false,
      "persistent": true
    },
...
```

It's unclear when the condition will or won't load it

**Comment by xbtmatt** [comment]
Wish stuff like this was in the release log changes, the most I could find was a reference to the big PR that changed it, but no details on how it actually changed

**Reply by anthonyshew** [reply]
This change is listed prominently in several places:
- The release blog post: https://turbo.build/blog/turbo-2-0#breaking-changes
- The upgrade guide: https://turbo.build/repo/docs/crafting-your-repository/upgrading#update-turbo-run-commands
- The release notes: https://github.com/vercel/turbo/releases/tag/v2.0.0

**Reply by xbtmatt** [reply]
thanks! I just thought a link to the upgrade guide in the release notes at the top would be nice. Appreciate the links though!

**Reply by anthonyshew** [reply]
That's an outstanding call. Just added that now. Thank you for your wisdom. 🧠

**Reply by xbtmatt** [reply]
😄🔥 next.js and turbo have some of the best and most extensive docs I've ever seen, so maybe I've just been spoiled!

**Comment by ya2s** [comment]
> Also, when I run yarn dev, it seems that I am able to read the .env environment variables even though they are not set.

https://turbo.build/repo/docs/crafting-your-repository/using-environment-variables#avoid-creating-or-mutating-environment-variables-at-runtime

**Reply by anthonyshew** [reply]
Added some clarification here: https://turbo.build/repo/docs/crafting-your-repository/using-environment-variables#handling-env-files

**Comment by ya2s** [comment]
i have created an environment that can achieve the minimum configuration.
https://github.com/yamatsum/turbo-v2-poc/pull/1
